src/command/defaults/BanCommand.php

```diff
@@ -25,42 +25,38 @@
 
 use pocketmine\command\Command;
 use pocketmine\command\CommandSender;
-use pocketmine\command\utils\InvalidCommandSyntaxException;
+use pocketmine\command\overload\CommandOverload;
+use pocketmine\command\overload\RawParameter;
+use pocketmine\command\overload\StringParameter;
 use pocketmine\lang\KnownTranslationFactory;
 use pocketmine\permission\DefaultPermissionNames;
 use pocketmine\player\Player;
-use function array_shift;
-use function count;
-use function implode;
 
-class BanCommand extends VanillaCommand{
+class BanCommand extends Command{
 
 	public function __construct(string $namespace, string $name){
 		parent::__construct(
 			$namespace,
 			$name,
+			[new CommandOverload(
+				[
+					new StringParameter("playerName", "player name"),
+					new RawParameter("reason", "reason")
+				],
+				DefaultPermissionNames::COMMAND_BAN_PLAYER,
+				self::execute(...)
+			)],
 			KnownTranslationFactory::pocketmine_command_ban_player_description(),
-			KnownTranslationFactory::commands_ban_usage()
 		);
-		$this->setPermission(DefaultPermissionNames::COMMAND_BAN_PLAYER);
 	}
 
-	public function execute(CommandSender $sender, string $commandLabel, array $args){
-		if(count($args) === 0){
-			throw new InvalidCommandSyntaxException();
-		}
-
-		$name = array_shift($args);
-		$reason = implode(" ", $args);
+	private static function execute(CommandSender $sender, string $playerName, string $reason) : void{
```

Comment:
I know this is legacy BC compat, but is there a way we can implement a player argument here now? It shouldn't take much more effort.